File: app/src/main/java/com/example/javademogithubpractice/network/model/AuthRequestModel.java


package com.example.javademogithubpractice.network.model;

import com.example.javademogithubpractice.AppConfig;
import com.example.javademogithubpractice.BuildConfig;
import com.google.gson.annotations.SerializedName;

import java.util.Arrays;
import java.util.List;



public class AuthRequestModel {

    private List<String> scopes;
    private String note;
    private String noteUrl;
    @SerializedName("client_id") private String clientId;
    @SerializedName("client_secret") private String clientSecret;

    public static AuthRequestModel generate(){
        AuthRequestModel model = new AuthRequestModel();
        model.scopes = Arrays.asList("user", "repo", "gist", "notifications");
        model.note = BuildConfig.APPLICATION_ID;
        model.clientId = AppConfig.DEMOGITHUB_CLIENT_ID;
        model.clientSecret = AppConfig.DEMOGITHUB_CLIENT_SECRET;
        model.noteUrl = AppConfig.REDIRECT_URL;
        return model;
    }

    public List<String> getScopes() {
        return scopes;
    }

    public String getNote() {
        return note;
    }

    public String getNoteUrl() {
        return noteUrl;
    }

    public String getClientId() {
        return clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/http/model/AuthRequestModel.kt
package com.example.kotlindemogithubproject.http.model

import com.example.kotlindemogithubproject.AppConfig
import com.example.kotlindemogithubproject.BuildConfig
import com.google.gson.annotations.SerializedName

import java.util.Arrays


class AuthRequestModel {
    var scopes: List<String>? = null
        private set
    var note: String? = null
        private set
    var noteUrl: String? = null
        private set
    @SerializedName("client_id")
    var clientId: String? = null
        private set
    @SerializedName("client_secret")
    var clientSecret: String? = null
        private set

    companion object {
        fun generate(): AuthRequestModel {
            val model = AuthRequestModel()
            model.scopes = Arrays.asList("user", "repo", "gist", "notifications")
            model.note = BuildConfig.APPLICATION_ID
            model.clientId = AppConfig.OPENHUB_CLIENT_ID
            model.clientSecret = AppConfig.OPENHUB_CLIENT_SECRET
            model.noteUrl = AppConfig.REDIRECT_URL
            return model
        }
    }
}
